source/agora/node/Ledger.d

```diff
@@ -232,21 +233,10 @@ unittest
     ledger.makeBlock();
     assert(ledger.getLastBlock().txs[$-1] == tx);
 
-    // getLastBlock Testing serialization
-    // Compare the serialization hexstring with the origin Ledger data.
-    const ubyte[] data = serializeFull(ledger.getLastBlock());
-
-    const string serializeData =
-     "02A91D85E7279998D31FE887281D0031D8C19FA732DC0BAC3A323569B19A14AD4B274255397A9A56"
-    ~"F14930CFAF7C7648377602EF0EF0E01821EE18F676092D060100000000000000DD9AAF1064DA6746"
-    ~"2DC1CC496E0374084058403429A1F74B72663F6CCA35B80AECD85063383D372F9DBCD770EF654B96"
-    ~"99F8C8801E6FD8041E9F517DF0B7CD2DE8D87AE11BDB8A861F7B0BD6EE6D7D37657FDD47CF8ED340"
-    ~"CDE27387366F9A64ABC91788162EFE5B3C9DAA4104113445994C7F0497199FF442E88F78C582A2CF"
-    ~"0000000066D002DC3D4F02352BF1E9478501CCF78FEA23AAFBAD4652630297C030A2F53E028BE2CD"
-    ~"CEEBADCCAD09908A375495919B49F263127B9BCC06DC8519A51EF80D005A6202000000009D0238E0"
-    ~"A171400BC6D68A9D9B316ACD5109649113A05C284F4296D2B30122F5";
-
-    assert(data.toHexString() == serializeData);
+    // getLastBlock Testing serialization & deserialization
+    // Compare the serialization data with the origin Ledger data.
+    const ubyte[] serializedData = serializeFull(ledger.getLastBlock());
+    assert(ledger.getLastBlock == deserializeBlock(serializedData));
```

Comment:
Nice!

Reply:
I would suggest moving this test somewhere else, like in `Serialize.d` or `Deserialize.d`. It's only serializing a block, it doesn't need the Ledger class for anything here. You can make your own block in a separate unittest.

Reply:
I will proceed with the test code on my own in the next PR  :)

Reply:
Move to Deserializer